neopixel.py

```diff
@@ -85,23 +95,33 @@ def __init__(self, num_leds, state_machine, pin, mode="RGB", delay=0.0001):
         :param delay: [default: 0.0001] delay used for latching of leds when sending data
         """
         self.pixels = array.array("I", [0] * num_leds)
+        # self.pixels_out = array.array("I", [0] * num_leds)
         self.mode = mode
         self.W_in_mode = 'W' in mode
+        bpp = 3
         if self.W_in_mode:
             # RGBW uses different PIO state machine configuration
-            self.sm = rp2.StateMachine(state_machine, sk6812, freq=8000000, sideset_base=Pin(pin))
+            self.sm = rp2.StateMachine(state_machine, sk6812, freq=12_800_000, sideset_base=Pin(pin))
             # tuple of values required to shift bit into position (check class desc.)
             self.shift = ((mode.index('R') ^ 3) * 8, (mode.index('G') ^ 3) * 8,
                           (mode.index('B') ^ 3) * 8, (mode.index('W') ^ 3) * 8)
+            bpp = 4
         else:
-            self.sm = rp2.StateMachine(state_machine, ws2812, freq=8000000, sideset_base=Pin(pin))
+            self.sm = rp2.StateMachine(state_machine, ws2812, freq=12_800_000, sideset_base=Pin(pin))
```

Comment:
Shouldn't this also have the new PIO code as only the sk6812 program is changed? 

If I understand this correctly, the program is now same for both strips, so maybe we can have just one program (but the pull_threshold will need to be adjusted for each state machine like in https://learn.adafruit.com/intro-to-rp2040-pio-with-circuitpython/advanced-using-pio-to-drive-neopixels-in-the-background).

A comment with link to original version this is based on should probably be included as well, to credit the authors of the original work.